app/src/org/commcare/views/widgets/QuestionWidget.java

```diff
@@ -601,13 +604,31 @@ public FormIndex getFormId(){
         return mPrompt.getIndex();
     }
 
-    public void setChangedListener(WidgetChangedListener wcl){
+    public void setChangedListeners(WidgetChangedListener wcl,
+                                    BlockingActionsManager blockingActionsManager){
         widgetChangedListener = wcl;
+        this.blockingActionsManager = blockingActionsManager;
     }
 
+    protected void fireDelayed(DelayedBlockingAction delayedBlockingAction) {
+        if(this.blockingActionsManager != null) {
+            blockingActionsManager.queue(delayedBlockingAction);
+        }
+    }
+
+    protected void widgetEntryChangedDelayed() {
+        fireDelayed(new DelayedBlockingAction(System.identityHashCode(this), 400) {
```

Comment:
Hm, a reasonable point. Mostly just dislike keeping around raw object references when we don't have to (to prevent accidental memory leaks, etc). but does seem a bit convoluted.